plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2215,9 +2212,7 @@ private TableStatisticsMetadata getStatisticsCollectionMetadata(
                 .filter(columnMetadata -> analyzeColumnNames.contains(columnMetadata.getName()))
                 .forEach(columnMetadata -> {
                     if (!(columnMetadata.getType() instanceof FixedWidthType)) {
-                        if (existingStatistics.isEmpty() || totalSizeStatisticsExists(existingStatistics.get().getColumnStatistics(), columnMetadata.getName())) {
```

Comment:
Check for previous statistics of total size is probably redundant. If column was not selected to ANALYZE it should be already filtered by `.filter(columnMetadata -> analyzeColumnNames.contains(columnMetadata.getName()))`
I could not find any other case when those statistics are not present except for case when new column is added.

Reply:
The other situation would be if a table was analyzed before we collected `TOTAL_SIZE_IN_BYTES`, upgraded Trino, and then analyzed the table again/did an insert.

That was added in 387 (June 2022), @ebyhr @findepi think that's long enough?

Reply:
good point, Alex. let's assume this is fine.